### PR TITLE
Add support for custom httpx client configuration in AzureOpenAI

### LINKED ISSUE #8875: [Feature Request]: Support for Custom httpx Client Configuration in AzureOpenAI
### Feature Description

**Description:**
I am proposing an enhancement to the AzureOpenAI class within the llama_index library to support custom httpx client configurations. This feature would allow users to specify their own httpx.Client instance when creating an AzureOpenAI object. This is particularly useful for scenarios where specific client configurations are required, such as custom timeouts, SSL certificates, or other HTTP client settings.

**Suggested Implementation:**
The enhancement could be implemented by adding an optional parameter to the AzureOpenAI constructor (and possibly other relevant methods) that accepts an httpx.Client instance. This client would then be used internally for all HTTP requests made by the AzureOpenAI instance.

Example:
```import httpx
from llama_index.llms import AzureOpenAI

custom_httpx_client = httpx.Client(...)  # Custom configuration
azure_ai = AzureOpenAI(http_client=custom_httpx_client, ...)```

### Reason

Currently, LlamaIndex's AzureOpenAI class does not expose a way to inject or modify the underlying httpx.Client used for HTTP requests. This limitation prevents users from configuring custom client settings such as SSL certificates, timeouts, or other specific HTTP client behaviors. Existing approaches, like directly modifying the private _client attribute (monkey patching), are not ideal as they rely on altering internal, private components of the class, leading to potential maintenance and compatibility issues with future updates of the library. A formal mechanism to set a custom httpx.Client would provide a more robust and maintainable solution.

### Value of Feature

This feature adds significant value by enhancing flexibility, allowing users to tailor the HTTP client to specific needs, such as security configurations and network requirements. It promotes better integration with diverse environments and ensures that the library can be more widely adopted in various secure and customized network setups.

### PR DESCRIPTION
**Pull request by aaronjimv**
# Description

This PR includes the changes requested in **[Feature Request]: Support for Custom httpx Client Configuration in AzureOpenAI #8875**

A new attribute, `http_client` , was added to `azure_openai.py` constructor . If a custom one is not provided, a default httpx client will be created.

Fixes #8875 

## Type of Change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by aaronjimv** [comment]
Hello, I am open to any feedback or suggestions.

File: llama_index/llms/azure_openai.py [hunk 1]
```diff
@@ -88,6 +91,10 @@ def __init__(
             "azure_endpoint", azure_endpoint, "AZURE_OPENAI_ENDPOINT", ""
         )
 
+        # Use the custom httpx client if provided.
+        # Otherwise a default one will be created.
+        self._http_client = http_client or httpx.Client()
```
Comment (logan-markewich) [inline llama_index/llms/azure_openai.py:96]: would it be better to leave it as `None` and let the `openai` package handle creating the client?

Comment (aaronjimv) [reply llama_index/llms/azure_openai.py:96]: Ok, I understand👍

Comment (theTechie) [reply llama_index/llms/azure_openai.py]: This seems to be causing a problem in 0.9.1? #8944

Comment (aaronjimv) [reply llama_index/llms/azure_openai.py]: Hi, thank for notify me. I think that the fixing is in #8955

**Pull request merged**